device/core/src/client.ts

```diff
@@ -1,885 +0,0 @@
-   * @param {Function}  Transport     A transport constructor.
-   *
-   * @throws {ReferenceError}         If the connStr parameter is falsy.
-   *
-   * @returns {module:azure-iothub.Client}
-   */
-  static fromSharedAccessSignature(sharedAccessSignature: string, transportCtor: any): Client {
-    /*Codes_SRS_NODE_DEVICE_CLIENT_16_029: [The fromSharedAccessSignature method shall throw a ReferenceError if the sharedAccessSignature argument is falsy.] */
-    if (!sharedAccessSignature) throw new ReferenceError('sharedAccessSignature is \'' + sharedAccessSignature + '\'');
-
-    /*Codes_SRS_NODE_DEVICE_CLIENT_16_088: [The `fromSharedAccessSignature` method shall create a new `SharedAccessSignatureAuthorizationProvider` object with the shared access signature passed as argument, and pass this object to the transport constructor.]*/
-    const authenticationProvider = SharedAccessSignatureAuthenticationProvider.fromSharedAccessSignature(sharedAccessSignature);
-
-    /*Codes_SRS_NODE_DEVICE_CLIENT_16_030: [The fromSharedAccessSignature method shall return a new instance of the Client object] */
-    return new Client(new transportCtor(authenticationProvider), null, new BlobUploadClient(authenticationProvider));
-  }
-
-  /**
-   * @method                        module:azure-iot-device.Client.fromAuthenticationMethod
-   * @description                   Creates an IoT Hub device client from the given authentication method and using the given transport type.
-   * @param authenticationProvider  Object used to obtain the authentication parameters for the IoT hub.
-   * @param transportCtor           Transport protocol used to connect to IoT hub.
-   */
-  static fromAuthenticationProvider(authenticationProvider: AuthenticationProvider, transportCtor: any): Client {
-    /*Codes_SRS_NODE_DEVICE_CLIENT_16_089: [The `fromAuthenticationProvider` method shall throw a `ReferenceError` if the `authenticationProvider` argument is falsy.]*/
-    if (!authenticationProvider) {
-      throw new ReferenceError('authenticationMethod cannot be \'' + authenticationProvider + '\'');
-    }
-
-    /*Codes_SRS_NODE_DEVICE_CLIENT_16_092: [The `fromAuthenticationProvider` method shall throw a `ReferenceError` if the `transportCtor` argument is falsy.]*/
-    if (!transportCtor) {
-      throw new ReferenceError('transportCtor cannot be \'' + transportCtor + '\'');
-    }
-
-    /*Codes_SRS_NODE_DEVICE_CLIENT_16_090: [The `fromAuthenticationProvider` method shall pass the `authenticationProvider` object passed as argument to the transport constructor.]*/
-    /*Codes_SRS_NODE_DEVICE_CLIENT_16_091: [The `fromAuthenticationProvider` method shall return a `Client` object configured with a new instance of a transport created using the `transportCtor` argument.]*/
-    return new Client(new transportCtor(authenticationProvider), null, new BlobUploadClient(authenticationProvider));
-  }
-}
-
-export namespace Client {
```

Comment:
>Client [](start = 17, length = 6)

it looks like we're losing the content of this namespace which was exported as part of the public API of th package (i believe). At least from a typescript perspective this would be considered a breaking change (customers using the content of this namespace would lose some type definitions). I think this needs to be brought back somewhere.